connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/expressions/UserDefinedFunction.scala

```diff
@@ -142,7 +144,15 @@ object ScalarUserDefinedFunction {
 
     ScalarUserDefinedFunction(
       function = function,
-      inputEncoders = parameterTypes.map(tag => ScalaReflection.encoderFor(tag)),
+      // Input can be a row because the input data schema can be found from the plan.
+      inputEncoders = parameterTypes.map(tag =>
```

Comment:
Can you just add this to `ScalaReflection`?

Reply:
Moved. Need to add a new method as not all places the Row encoder conversion should be supported. e.g. The client UDF return encoder cannot be a Row encoder.